sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManagerBase.scala

```diff
@@ -63,16 +74,340 @@ abstract class GpuShuffleBlockResolverBase(
   override def stop(): Unit = wrapped.stop()
 }
 
+/**
+ * The `ShuffleWriteMetricsReporter` is based on accumulators, which are not thread safe.
+ * This class is a thin wrapper that adds synchronization, since these metrics will be written
+ * by multiple threads.
+ * @param wrapped
+ */
+class ThreadSafeShuffleWriteMetricsReporter(wrapped: ShuffleWriteMetricsReporter)
+  extends ShuffleWriteMetrics {
+  override private[spark] def incBytesWritten(v: Long): Unit = synchronized {
```

Comment:
I think most of these are updated in batches, just curious if this synchronization causes us much time waiting?  might be something to look at later if you haven't

Reply:
I haven't looked at this. I kind of ran into this late where the tests from spark were actually testing the metrics. Yes absolutely we could do better than this, but I figure that could be a punt.